dips/search_helpers.py

```diff
@@ -0,0 +1,124 @@
+    """
+    search.aggs.bucket('formats', 'terms', field='fileformat.raw', size=10000)
+    # The collections agg. is made directly over the title to avoid hitting
+    # the ORM to get the title with the id or to use a composite agg. with
+    # multiple sources. The biggest downside is that the titles are used as
+    # value in the request parameters, which can create long URLs as multiple
+    # selection is allowed. This agg. is optional and not added in DIP pages.
+    if collections:
+        search.aggs.bucket(
+            'collections', 'terms', field='collection.title.raw', size=10000)
+
+    return search
+
+
+def get_and_apply_digital_file_filters(request, search):
+    filters = {'formats': request.GET.getlist('for', []),
+               'collections': request.GET.getlist('col', []),
+               'start_date': request.GET.get('start_date', ''),
+               'end_date': request.GET.get('end_date', '')}
+    if filters['formats']:
+        search = search.query('terms', **{'fileformat.raw': filters['formats']})
+    if filters['collections']:
+        search = search.query(
+            'terms', **{'collection.title.raw': filters['collections']})
+    if filters['start_date']:
+        try:
+            datetime.strptime(filters['start_date'], '%Y-%m-%d')
+            search = search.query(
+                'range', **{'datemodified': {
+                    'gte': filters['start_date'],
+                    'format': 'yyyy-MM-dd'
+                }})
+        except ValueError:
+            messages.error(request, _(
+                'Incorrect date format for start date (%(date)s). '
+                'Expected: yyyy-mm-dd.' % {'date': filters['start_date']}
+            ))
+    if filters['end_date']:
+        try:
+            datetime.strptime(filters['end_date'], '%Y-%m-%d')
+            search = search.query(
+                'range', **{'datemodified': {
+                    'lte': filters['end_date'],
+                    'format': 'yyyy-MM-dd'
+                }})
+        except ValueError:
+            messages.error(request, _(
+                'Incorrect date format for end date (%(date)s). '
+                'Expected: yyyy-mm-dd.' % {'date': filters['end_date']}
+            ))
+
+    return (search, filters)
+
+
+def get_page_from_search(search, request):
+    """
+    Create paginator and return current page based on the current search
+    and the page and limit request parameters. Limit defaults to 10 and
+    can't be set over 100.
+    """
+    try:
+        limit = int(request.GET.get('limit', 10))
+        if limit <= 0 or limit > 100:
+            raise ValueError
+    except ValueError:
+        limit = 10
+
+    paginator = Paginator(search, limit)
+    page_no = request.GET.get('page')
+    try:
+        page = paginator.page(page_no)
+    except PageNotAnInteger:
+        page = paginator.page(1)
+    except EmptyPage:
+        page = paginator.page(paginator.num_pages)
+
+    return page
```

Comment:
Sorry, this file was an initial implementation that was committed by mistake and removed in the latest commit.